File: tests/nodejs/cases/case_277_interface_sink_bridge_07.js
/**
 * Task case 277: interface_sink_bridge_07 deep stress pattern.
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case277InterfaceSinkBridge07(input) {
  const taskName = 'interface_sink_bridge_07';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'extreme',
    stage: 'stress',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  const sink = { put(obj) { retainedAudit.push(obj); } };
  // ESCAPE: object method bridge to retained sink.
  sink.put(payload);
  return 'ok';
}

module.exports = { case277InterfaceSinkBridge07 };

File: tests/go/cases/case_277_interface_sink_bridge_07.go
package escape_tests

var retainedCase277 = []map[string]string{}

type case277Sinker interface{ Put(map[string]string) }
type case277Bridge struct{}

func (case277Bridge) Put(v map[string]string) {
	retainedCase277 = append(retainedCase277, v)
}

func Case277InterfaceSinkBridge07(input string) string {
	raw := input
	if raw == "" {
		raw = "sample"
	}
	payload := map[string]string{
		"task": "interface_sink_bridge_07",
		"entity": "extreme",
		"stage": "stress",
		"input": raw,
		"checksum": "interface_sink_bridge_07:" + raw,
	}
	var s case277Sinker = case277Bridge{}
	// ESCAPE: interface bridge dispatch to sink.
	s.Put(payload)
	return "ok"
}
